UnityProject/Assets/Scripts/Managers/ComponentManager.cs

```diff
@@ -43,15 +64,24 @@ public static bool TryGetUniversalObjectPhysics(GameObject gameObject, out Unive
 	private void OnEnable()
 	{
 		SceneManager.activeSceneChanged += OnRoundRestart;
+		EventManager.AddHandler(Event.RoundEnded, OnRoundEnded);
 	}
 
 	private void OnDisable()
 	{
 		SceneManager.activeSceneChanged -= OnRoundRestart;
+		EventManager.RemoveHandler(Event.RoundEnded, OnRoundEnded);
 	}
 
 	void OnRoundRestart(Scene oldScene, Scene newScene)
 	{
 		ObjectToPhysics.Clear();
+		ObjectToCommonComponent.Clear();
+	}
+
+	void OnRoundEnded()
```

Comment:
```suggestion
	private void OnRoundEnded()
```

add the explicit access modifier

Reply:
this will get modified by 6opoDuJIo's work